### PR TITLE
Use PX4 Board information metadata to identify board IDs

### PR DESCRIPTION
**Pull request by junwoo091400**
## Context
This is the QGC part of the PR: https://github.com/PX4/PX4-Autopilot/pull/19848
(To be written)

### CODE REVIEW

**Comment by junwoo091400** [comment]
Status updated in PX4 side: https://github.com/PX4/PX4-Autopilot/pull/19848#issuecomment-1191590296

**Comment by junwoo091400** [comment]
The PR diff looks big but actually the majority is from the Manifest file (1k+ lines added) :rofl:

File: src/VehicleSetup/FirmwareUpgradeController.cc [hunk 1]
```diff
@@ -646,73 +646,72 @@ void FirmwareUpgradeController::_downloadPX4Manifest(void)
...
-            const QJsonObject& boardInfoUnitJson = boardInfoArray[board_idx].toObject();
+    // Read in board informations of each board target
+    QJsonArray boardInfoArray = json[_px4ManifestBoardInfoJsonKey].toArray();
 
-            // Add the basic board information
-            boardInfoUnit.boardName = boardInfoUnitJson[_px4ManifestBoardNameJsonKey].toString();
-            boardInfoUnit.targetName = boardInfoUnitJson[_px4ManifestTargetNameJsonKey].toString();
-            boardInfoUnit.description = boardInfoUnitJson[_px4ManifestDescriptionJsonKey].toString();
-            boardInfoUnit.boardID = boardInfoUnitJson[_px4ManifestBoardIDJsonKey].toInt();
+    for (int board_idx = 0; board_idx < boardInfoArray.count(); board_idx++) {
```
Comment (patrickelectric) [inline src/VehicleSetup/FirmwareUpgradeController.cc:681]: `for(const auto& board: boardInfoArray)`

File: src/VehicleSetup/FirmwareUpgradeController.cc [hunk 2]
```diff
@@ -646,73 +646,72 @@ void FirmwareUpgradeController::_downloadPX4Manifest(void)
...
 
-            // Add optional USB Autoconnect info
-            boardInfoUnit.productID = boardInfoUnitJson[_px4ManifestProductIDJsonKey].toInt();
-            qInfo() << "Adding Product ID " << boardInfoUnit.productID << " to board " << boardInfoUnit.boardName;
-            boardInfoUnit.productName = boardInfoUnitJson[_px4ManifestProductNameJsonKey].toString();
-            boardInfoUnit.vendorID = boardInfoUnitJson[_px4ManifestVendorIDJsonKey].toInt();
-            boardInfoUnit.vendorName = boardInfoUnitJson[_px4ManifestVendorNameJsonKey].toString();
+        // Add in the list of build variants
+        QJsonArray buildVariantsArray = boardInfoUnitJson[_px4ManifestBuildVariantsJsonKey].toArray();
+        for (int build_variant_idx = 0; build_variant_idx < buildVariantsArray.count(); build_variant_idx++) {
```
Comment (patrickelectric) [inline src/VehicleSetup/FirmwareUpgradeController.cc:693]: `for(const auto& build_variant: buildVariantsArray)`

**Review by patrickelectric — CHANGES_REQUESTED** [review]
I did a quick review, the patch is a bit big, but for general tips, some commits can be squashed and some of the code can be improved.

**Comment by DonLakeFlyer** [comment]
Is this still being worked on?

**Comment by DonLakeFlyer** [comment]
Not supported by mavlnk

**Pull request closed without merging**